cns/cnireconciler/podinfoprovider.go

```diff
@@ -44,3 +67,32 @@ func cniStateToPodInfoByIP(state *api.AzureCNIState) (map[string]cns.PodInfo, er
 	}
 	return podInfoByIP, nil
 }
+
+func endpointStateToPodInfoByIP(state map[string]*restserver.EndpointInfo) (map[string]cns.PodInfo, error) {
+	podInfoByIP := map[string]cns.PodInfo{}
+	for containerID, endpointInfo := range state {
+		for ifname, ipinfo := range endpointInfo.IfnameToIPMap {
+			if _, ok := podInfoByIP[ipinfo.IPv4.IP.String()]; ok {
```

Comment:
its good to check ipinfo.IPv4!= nil as like ipv6. if in future, if azure supports single stack ipv6 , then this would be an issue